mpt_trie/src/nibbles.rs

```diff
@@ -119,7 +174,73 @@ impl_to_nibbles!(u32);
 impl_to_nibbles!(u64);
 impl_to_nibbles!(U128);
 impl_to_nibbles!(U256);
-impl_to_nibbles!(U512);
+impl_to_nibbles!(NibblesIntern);
+
+impl<'a> TryFrom<&'a Nibbles> for U256 {
+    type Error = NibblesToTypeError;
+
+    fn try_from(value: &'a Nibbles) -> Result<U256, NibblesToTypeError> {
+        let NibblesIntern(ref arr) = value.packed;
+        if arr[4] != 0 {
+            return Err(NibblesToTypeError::Overflow);
+        }
+
+        let mut ret = [0; 4];
```

Comment:
I feel `let ret = [arr[0], arr[1], arr[2], arr[3]];` is more readable. Plus less `mut` which makes me happy.